### PR TITLE
Remove PG_INDEXER feature flag and drop obsolete code

### PR DESCRIPTION
**Pull request by habdelra**

### CODE REVIEW

File: packages/runtime-common/url-map.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// TODO we don't need this anymore
```
Comment (habdelra) [inline packages/runtime-common/url-map.ts]: don't forget this

File: packages/runtime-common/realm.ts [hunk 2]
```diff
@@ -883,10 +778,7 @@ export class Realm {
     try {
       // local requests are allowed to query the realm as the index is being built up
       if (!isLocal) {
-        // allow any WIP index requests to query the index while it's building up
-        if (!request.headers.get('X-Boxel-Use-WIP-Index')) {
-          await this.ready;
-        }
```
Comment (habdelra) [inline packages/runtime-common/realm.ts:894]: turns out I was using the loader wrong. the realm's loader has the internal fetch URL handler. once I started passing a clone of that to the worker then this condition became unnecessary

File: packages/runtime-common/indexer.ts [hunk 3]
```diff
@@ -1052,21 +1063,48 @@ export class Batch {
...
+            // than any unfinished work. However, this might mask actual
+            // indexing bugs. Having the new generation logic more sensitive to
+            // unfinished indexing work should hopefully bring to light any
+            // future indexing bugs more readily.
+            `${message}. created a new generation while there was still unfinished indexing. ` +
+            `The most likely reason this happens is that there was an error encountered during incremental ` +
+            `indexing that prevented the indexing from completing (and realm version increasing), ` +
+            `then the realm was restarted and the left over WIP indexing artifact(s) collided with the ` +
+            `from-scratch indexing. To resolve this issue delete the WIP indexing artifacts (the row(s) ` +
+            `that triggered the unique constraint) and restart the realm.`;
```
Comment (habdelra) [inline packages/runtime-common/indexer.ts:1101]: trying to win the oscar for best error message :laughing:

File: packages/realm-server/main.ts [hunk 4]
```diff
@@ -199,30 +190,31 @@ if (distURL) {
...
-              indexRunner: getRunner,
-            });
-            await worker.run();
-          }
+          // Note for future: we are taking advantage of the fact that the realm
+          // does not need to auth with itself and are passing in the realm's
+          // loader which includes a url handler for internal requests that
+          // bypasses auth. when workers are moved outside of the realm server
+          // they will need to provide realm authentication credentials when
+          // indexing.0
```
Comment (backspace) [inline packages/realm-server/main.ts]:
```suggestion
          // indexing.
```

File: packages/runtime-common/indexer.ts (hunk 3 above)
Comment (backspace) [reply packages/runtime-common/indexer.ts:1101]: seems helpful!

File: packages/runtime-common/realm.ts [hunk 5]
```diff
@@ -937,6 +829,9 @@ export class Realm {
     return undefined;
   }
 
+  // TODO we could really improve performance if this utilized the index instead
+  // of directly hitting the filesystem--especially the TS transpilation
+  // involved in making JS.
```
Comment (backspace) [inline packages/runtime-common/realm.ts:839]: is this a full project or could an issue be opened for it?

**Review by backspace — APPROVED** [review]
seems like a smooth transition!

**Comment by backspace** [comment]
hmm actually I approved before I saw [these `host` test failures](https://github.com/cardstack/boxel/runs/24836005221).

**Comment by habdelra** [comment]
> hmm actually I approved before I saw [these `host` test failures](https://github.com/cardstack/boxel/runs/24836005221).

No prob, that’s my Monday work

File: packages/runtime-common/realm.ts (hunk 5 above)
Comment (habdelra) [reply packages/runtime-common/realm.ts:839]: I think it’s probably a project

File: packages/runtime-common/indexer.ts (hunk 3 above)
Comment (jurgenwerk) [reply packages/runtime-common/indexer.ts:1101]: 🥇

**Review by jurgenwerk — APPROVED** [review]
🎉

**Pull request merged**